tests/console/validate_encrypt.pm

```diff
@@ -19,30 +19,86 @@ use strict;
 
-sub run {
+sub parse_crypttab {
+    my @lines    = split(/\n/, script_output("cat /etc/crypttab"));
+    my $crypttab = {};
+    foreach (@lines) {
+        next if /^\s*#.*$/;
+        if ($_ =~ /(?<name>.+?)\s+(?<encrypted_device>.+?)($|\s+(?<password>.*?)($|\s+(?<options>.*)))/) {
+            $crypttab->{$+{name}} = {
+                encrypted_device => $+{encrypted_device},
+                password         => $+{password},
+                options          => $+{options}};
+        }
+    }
+    return $crypttab;
+}
 
-    record_info('crypttab file', 'Verify whether \'/etc/crypttab\' (encrypted device table) file exists.');
-    assert_script_run('test -f /etc/crypttab', fail_message => 'No /etc/crypttab found!');
+sub parse_cryptsetup_status {
+    my $dev    = shift;
+    my @lines  = split(/\n/, script_output("cryptsetup status $dev"));
+    my $status = {message => shift @lines};
+    foreach (@lines) {
+        if ($_ =~ /\s+(?<param>.*):\s+(?<value>.*)$/) {
+            my ($param, $value) = ($+{param}, $+{value});
+            $param =~ s/ /_/g;
+            $status->{$param} = $value;
+        }
+    }
+    return $status;
+}
 
-    record_info('crypt vol status', 'Verify the crypted volumes are active');
-    my @crypted_volumes = script_output q[cat /etc/crypttab | awk '{print $1}'];
-    foreach (@crypted_volumes) {
-        assert_script_run qq[cryptsetup status $_ | grep "is active"];
+sub verify_crypttab {
+    my $crypttab_expected = shift;
+    record_info("crypttab", "Verify file existence and content");
+    assert_script_run("test -f /etc/crypttab", fail_message => "No /etc/crypttab found");
+    my $crypttab = parse_crypttab();
+    foreach (@{$crypttab_expected->{devices}}) {
+        die "Device $_ not found in /etc/crypttab" unless exists($crypttab->{$_});
+        delete $crypttab->{$_};
+    }
+    if (%{$crypttab}) {
+        die "/etc/crypttab contains more devices than expected:\n" . Dumper($crypttab);
     }
+}
 
-    record_info('Verify LUKS', 'Verify binary backups of LUKS header and keyslot areas storing and restoring');
-    foreach (@crypted_volumes) {
-        my $bkp_file = '/root/bkp_luks_header';
-
-        my $device = script_output q[cat /etc/crypttab | awk '{print $2}'];
-        next if (script_run('cryptsetup -v isLuks ' . $device) != 0);
-        assert_script_run('cryptsetup -v luksUUID ' . $device);
-        assert_script_run('cryptsetup -v luksDump ' . $device);
-        assert_script_run('cryptsetup -v luksHeaderBackup ' . $device . ' --header-backup-file ' . $bkp_file);
-        validate_script_output("file $bkp_file", sub { m/\bLUKS\sencrypted\sfile\b/ });
-        assert_script_run('cryptsetup -v --batch-mode luksHeaderRestore ' . $device . ' --header-backup-file ' . $bkp_file);
+sub verify_cryptsetup_status {
+    my $status_expected = shift;
+    record_info("active volumes", "Verify crypted volumes are active");
+    foreach my $dev (sort keys %{$status_expected}) {
```

Comment:
As I also proposed above, maybe it is worth to use `sub arrays_differ` from utils.pm here? WDYT?